lib/implementation.js

```diff
@@ -696,7 +697,10 @@ module.exports = function(self, options) {
         // from the Apostrophe Express module to ensure that user
         // defined cookie settings are respected.
         let aposExpressModule = self.apos.modules['apostrophe-express'];
-        req.session.cookie = new ExpressSessionCookie(aposExpressModule.sessionOptions.cookie);
+
+        if (req.user || !aposExpressModule.options.session.disableAnonCsrf) {
```

Comment:
This method only fires when and if the user is actually following a locale-switching link with a cross-domain session token, which a logged-out visitor will not have. I don't think this change is necessary.

Reply:
Man! It's a good thing I enjoy being wrong. It fires on ordinary page loads. Hrrrrrrrrrm.

Reply:
(or rather, it fires on links that can be generated by ordinary page loads.)